src/settings.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  ISetting,
+  SettingType,
+} from "@rocket.chat/apps-engine/definition/settings";
+
+export enum AppSetting {
+  AsanaClientId = "asana_client_id",
+  AsanaClientSecret = "asana_client_secret",
+  AsanaRedirectUri = "asana_redirect_uri",
+  AsanaWebhookSecret = "asana_webhook_secret",
+  NotificationColor = "notification_color",
+  DefaultWorkspace = "default_workspace",
+}
+
+export const settings: Array<ISetting> = [
```

Comment:
I don't think this is required, extending the App's configuration with getOauth2ClientInstance.setup method automatically adds these fields.
Please update your apps engine version and follow latest docs.